### ISSUE TITLE
add support for blink.cmp

### ISSUE THREAD

**Issue by olivertzeng** [windwp/nvim-autopairs#477]
**Is your feature request related to a problem? Please describe.**
https://github.com/Saghen/blink.cmp there is a brand new completion plugin called blink.cmp

**Describe the solution you'd like**
add support for it

**Describe alternatives you've considered**
No

**Additional context**
none

**Comment by megalithic** [comment]
@olivertzeng have you found a workaround/solution for autopair completion for blink, by chance?

**Comment by smjonas** [comment]
I have found this: https://github.com/Saghen/blink.cmp/discussions/157

Seems to work fine for me

**Comment by jabdinghoff** [comment]
> I have found this: [Saghen/blink.cmp#157](https://github.com/Saghen/blink.cmp/discussions/157)
> 
> Working fine for me

what does the nvim-autopairs config need to look like to work with
`completion = { accept = { auto_brackets = { enabled = true } } }`?

**Comment by nasvillanueva** [comment]
Blink's auto brackets option works fine for me so far (or maybe I just don't know the difference with nvim-autopairs), but maybe it can work with

```lua
local cmp_autopairs = require("nvim-autopairs.completion.cmp")

require("blink-cmp").setup({
  keymap = {
    ["<CR>"] = {
      function(cmp) 
        return cmp.accept({
          callback = cmp_autopairs.on_confirm_done
        })
      end
    } 
  }
})
```

I had this on my nvim-cmp config before I switched to blink

```lua
local cmp_autopairs = require("nvim-autopairs.completion.cmp")
cmp.event:on("confirm_done", cmp_autopairs.on_confirm_done())
```

**Comment by llakala** [comment]
> Blink's auto brackets option works fine for me so far (or maybe I just don't know the difference with nvim-autopairs), but maybe it can work with
> 
> local cmp_autopairs = require("nvim-autopairs.completion.cmp")
> 
> require("blink-cmp").setup({
>   keymap = {
>     ["<CR>"] = {
>       function(cmp) 
>         return cmp.accept({
>           callback = cmp_autopairs.on_confirm_done
>         })
>       end
>     } 
>   }
> })
> 
> I had this on my nvim-cmp config before I switched to blink
> 
> local cmp_autopairs = require("nvim-autopairs.completion.cmp")
> cmp.event:on("confirm_done", cmp_autopairs.on_confirm_done())

Tried it - didn't have any luck.

**Comment by llakala** [comment]
I hate stalebots.

**Comment by olivertzeng** [comment]
no

**Comment by olivertzeng** [comment]
Just leaving a comment here that this is still an issue

**Comment by DerRockWolf** [comment]
lol how stalebot ignored your message 💀

**Comment by onion108** [comment]
<img width="264" height="138" alt="Image" src="https://github.com/user-attachments/assets/c1c4eaca-a863-4a17-bff8-a7fed883abf5" />
nvim-autopairs seems to be broken working with blink.cmp recently... garbage being outputted like this
(mini.pairs works surprisingly fine?)

**Comment by ZIDOUZI** [comment]
> <img alt="Image" width="264" height="138" src="https://private-user-images.githubusercontent.com/57032603/563949988-c1c4eaca-a863-4a17-bff8-a7fed883abf5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM3MDY4NTUsIm5iZiI6MTc3MzcwNjU1NSwicGF0aCI6Ii81NzAzMjYwMy81NjM5NDk5ODgtYzFjNGVhY2EtYTg2My00YTE3LWJmZjgtYTdmZWQ4ODNhYmY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzE3VDAwMTU1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkZWE1OWRhNDY4YjAxYmQ1Y2QwODJkYmZkMTNlNTVhMjAxOWY4NmMwY2FhMTg1NWFkMmU0OWQ1Njk5ZDRlYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2CEdV-a5_G9FgliJK9G_AMSAnolzDMUkojLHqaLarXM"> nvim-autopairs seems to be broken working with blink.cmp recently... garbage being outputted like this (mini.pairs works surprisingly fine?)

same on me. do you have a solution?

**Comment by onion108** [comment]
> > <img alt="Image" width="264" height="138" src="https://private-user-images.githubusercontent.com/57032603/563949988-c1c4eaca-a863-4a17-bff8-a7fed883abf5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM3MDY4NTUsIm5iZiI6MTc3MzcwNjU1NSwicGF0aCI6Ii81NzAzMjYwMy81NjM5NDk5ODgtYzFjNGVhY2EtYTg2My00YTE3LWJmZjgtYTdmZWQ4ODNhYmY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzE3VDAwMTU1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkZWE1OWRhNDY4YjAxYmQ1Y2QwODJkYmZkMTNlNTVhMjAxOWY4NmMwY2FhMTg1NWFkMmU0OWQ1Njk5ZDRlYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2CEdV-a5_G9FgliJK9G_AMSAnolzDMUkojLHqaLarXM"> nvim-autopairs seems to be broken working with blink.cmp recently... garbage being outputted like this (mini.pairs works surprisingly fine?)
> 
> same on me. do you have a solution?

Nah. I tried to but I failed. So I just moved to `blink.pairs`, it's working pretty well at least for now.

**Comment by mesonoxianvlad** [comment]
I added the following keymap to blink that seems to be working ok so far.

```lua
      ["<CR>"] = {
        function(cmp)
          if cmp.is_visible() then
            return cmp.accept()
          else
            return require("nvim-autopairs").autopairs_cr()
          end
        end,
      },

```
